code/modules/transport/pods/MainWeapon.dm

```diff
@@ -380,6 +380,608 @@
+			if(istype(T,/turf/simulated/floor) && !istype(T,/turf/simulated/floor/airbridge) && !istype(T,/turf/simulated/floor/shuttle)\
+				&& !istype(T,/turf/simulated/floor/setpieces) && !istype(T,/turf/simulated/floor/martian))
+				return TRUE
+		//fallback: space is good
+		if (istype(T,/turf/space))
+			return TRUE
+
+	opencomputer(mob/user as mob)
+		if(user.loc != src.ship)
+			return
+		src.add_dialog(user)
+
+		var/dat = "<TT><B>Weapon Console</B><BR><HR>"
+		if(src.active)
+			dat +="<B>Weapon Mode:</B><BR>"
+			switch(mode)
+				if(EFIF_MODE_FLOORS)
+					dat+="<B>Floors</B><BR>"
+					dat+="<A href='?src=\ref[src];r_floors=1'>Switch to Reinforced Floors</A><BR>"
+					dat+="<A href='?src=\ref[src];walls=1'>Switch to Walls</A><BR>"
+					dat+="<A href='?src=\ref[src];repair=1'>Switch to Repair</A><BR>"
+					dat+="<A href='?src=\ref[src];cutter=1'>Switch to Disassembly</A><BR>"
+				if(EFIF_MODE_R_FLOORS)
+					dat+="<A href='?src=\ref[src];floors=1'>Switch to Floors</A><BR>"
+					dat+="<B>Reinforced Floors</B><BR>"
+					dat+="<A href='?src=\ref[src];walls=1'>Switch to Walls</A><BR>"
+					dat+="<A href='?src=\ref[src];repair=1'>Switch to Repair</A><BR>"
+					dat+="<A href='?src=\ref[src];cutter=1'>Switch to Disassembly</A><BR>"
+				if(EFIF_MODE_WALLS)
+					dat+="<A href='?src=\ref[src];floors=1'>Switch to Floors</A><BR>"
+					dat+="<A href='?src=\ref[src];r_floors=1'>Switch to Reinforced Floors</A><BR>"
+					dat+="<B>Walls</B><BR>"
+					dat+="<A href='?src=\ref[src];repair=1'>Switch to Repair</A><BR>"
+					dat+="<A href='?src=\ref[src];cutter=1'>Switch to Disassembly</A><BR>"
+				if(EFIF_MODE_REPAIR)
+					dat+="<A href='?src=\ref[src];floors=1'>Switch to Floors</A><BR>"
+					dat+="<A href='?src=\ref[src];r_floors=1'>Switch to Reinforced Floors</A><BR>"
+					dat+="<A href='?src=\ref[src];walls=1'>Switch to Walls</A><BR>"
+					dat+="<B>Repair</B><BR>"
+					dat+="<A href='?src=\ref[src];cutter=1'>Switch to Disassembly</A><BR>"
+				else
+					dat+="<A href='?src=\ref[src];floors=1'>Switch to Floors</A><BR>"
+					dat+="<A href='?src=\ref[src];r_floors=1'>Switch to Reinforced Floors</A><BR>"
+					dat+="<A href='?src=\ref[src];walls=1'>Switch to Walls</A><BR>"
+					dat+="<A href='?src=\ref[src];repair=1'>Switch to Repair</A><BR>"
+					dat+="<B>Disassembly</B><BR>"
+			dat+="<BR>"
+			dat+="Wide Field Mode <B>[src.wide_field ? "Active" : "Inactive"]</B> <A href='?src=\ref[src];wide_field=1'>(Toggle)</A><BR>"
+			dat+="[src.steel_sheets] of [src.max_sheets] Steel Sheets Loaded <A href='?src=\ref[src];load=1'>(Load)</A><BR>"
+
+		else
+			dat += {"<B><span style=\"color:red\">SYSTEM OFFLINE</span></B>"}
+		user.Browse(dat, "window=ship_main_weapon")
+		onclose(user, "ship_main_weapon")
+		return
+
+	Topic(href, href_list)
+		if(..())
+			return
+
+		if ((usr.contents.Find(src) || (in_interact_range(src, usr) && istype(src.loc, /turf))) || (issilicon(usr)))
+			src.add_dialog(usr)
```

Comment:
I don't think any of these are ever going to be true apart from the `issilicon` one and that probably isn't what you want?

Reply:
Cargo culted this one from the industrial utility arms. im shame

Removing it doesn't appear to have created any issues, so I'll probably do that.